Signal/src/util/OWSBackupExportJob.m

```diff
@@ -672,25 +853,34 @@ - (void)deleteRecordsFromCloud:(NSMutableArray<NSString *> *)obsoleteRecordNames
 
     if (obsoleteRecordNames.count < 1) {
         // No more records to delete; cleanup is complete.
-        completion(nil);
-        return;
+        return completion(nil);
+    }
+
+    if (self.isComplete) {
+        // Job was aborted.
+        return completion(nil);
     }
 
     CGFloat progress = (obsoleteRecordNames.count / (CGFloat)(obsoleteRecordNames.count + deletedCount));
     [self updateProgressWithDescription:NSLocalizedString(@"BACKUP_EXPORT_PHASE_CLEAN_UP",
                                             @"Indicates that the cloud is being cleaned up.")
                                progress:@(progress)];
 
-    NSString *recordName = obsoleteRecordNames.lastObject;
-    [obsoleteRecordNames removeLastObject];
+    static const NSUInteger kMaxBatchSize = 100;
+    NSMutableArray<NSString *> *batchRecordNames = [NSMutableArray new];
+    while (obsoleteRecordNames.count > 0 && batchRecordNames.count < kMaxBatchSize) {
+        NSString *recordName = obsoleteRecordNames.lastObject;
+        [obsoleteRecordNames removeLastObject];
+        [batchRecordNames addObject:recordName];
+    }
 
     __weak OWSBackupExportJob *weakSelf = self;
-    [OWSBackupAPI deleteRecordFromCloudWithRecordName:recordName
+    [OWSBackupAPI deleteRecordsFromCloudWithRecordNames:batchRecordNames
```

Comment:
The thing about CloudKit's "batch" methods is that although they offer tantalizing perf benefits, your code might become significantly more complicated to handle "partial failures."